src/ngraph/runtime/gpu/gpu_external_function.hpp

```diff
@@ -56,13 +57,14 @@ namespace ngraph
 
             public:
                 GPU_ExternalFunction(const std::shared_ptr<ngraph::Function>& function,
+                                     std::shared_ptr<GPU_Backend::BackendContext>& shared_context,
                                      bool release_function = true);
                 ~GPU_ExternalFunction();
                 std::shared_ptr<ngraph::runtime::gpu::GPU_CallFrame> make_call_frame();
                 std::unique_ptr<runtime::gpu::GPURuntimeContext>& ctx();
```

Comment:
I'm not sure if we should still have a public member function to get `ctx()` and `get_primitive_emitter()` since those are not own by `GPU_ExternalFunction` now.

Reply:
Agreed. This required a significant refactor of the runtime context use. Now the emitters maintain a non-owning pointer to the runtime context `ctx` which is given at construction from the BackendContext.